### PR TITLE
specs: change registration for Notify

### PR DESCRIPTION
**Pull request by Elyniss**

### CODE REVIEW

**Review by Cali93 — COMMENTED** [review]
`register` makes totally sense.
I just have some questions because we couple it with `enableSync`, wouldn't we face some edge cases ?
For example, a user registers and decides to not enable sync but wants to enable it later on. Would it be ok to call `register` again ?

**Comment by Elyniss** [comment]
>register makes totally sense.
I just have some questions because we couple it with enableSync, wouldn't we face some edge cases ?
For example, a user registers and decides to not enable sync but wants to enable it later on. Would it be ok to call register again ?

@Cali93 
Calling `register` again with `enableSync` param set to `true` should do the trick.

**Review by Cali93 — APPROVED** [review]

File: docs/specs/clients/notify/client-api.md [hunk 1]
```diff
@@ -44,13 +43,16 @@ abstract class Client {
...
-  public abstract enableSync({
+  // registers a blockchain account with an identity key if not yet registered on this client
+  // if enableSync is true then register sync keys
+  // returns the public identity key.
+  public abstract register(params: {
         account: string, 
-        onSign: (message: string) => Cacao.Signature
-  }): Promise<void>;
+        onSign: (message: string) => Cacao.Signature,
+        enableSync: boolean
```
Comment (Talhaali00) [inline docs/specs/clients/notify/client-api.md]: Should we just enable this by default?

**Review by TalhaAli00 — APPROVED** [review]

**Review by llbartekll — APPROVED** [review]

File: docs/specs/clients/notify/client-api.md (hunk 1 above)
Comment (Elyniss) [reply docs/specs/clients/notify/client-api.md]: I'm personally in favour of enabling Sync by default, and disallowing using Notify without Sync

Comment (pedrouid) [reply docs/specs/clients/notify/client-api.md]: Agreed. It should always be enabled. No need for this flag

Comment (Elyniss) [reply docs/specs/clients/notify/client-api.md]: Looks like we are all in agreement here. Flag removed

**Review by pedrouid — APPROVED** [review]
👍

**Review by devceline — APPROVED** [review]
LGTM

**Review by flypaper0 — APPROVED** [review]

**Pull request merged**